### PR TITLE
Configuration Service/API

### PR DESCRIPTION
**Pull request by drtyh2o**
This PR introduces a new API that allows loading of configuration settings required by applications. This API was written to be agnostic to the actual contents/format of the settings files. The environment variable `MDS_CONFIG_PATH` is set to the path of a folder containing one or more JSON settings files.

`GET /config/settings/foo` will load and return the contents of a `foo.json` file located in the config folder. `GET /config/settings` will load and return the contents of a `settings.json` file located in the config folder.

Also, there is a client stub that is used by the API, which can also be used directly by other services running in the cluster that have the local volume mounted and the environment variable set (e.g. event processor).

It is imagined that eventually this client stub may use `gRPC` or some other mechanism to communicate with a configuration service running independently in its own pod(s).

## Impacts
- [ ] Provider
- [ ] Agency
- [ ] Audit
- [ ] Policy
- [ ] Compliance
- [ ] Daily
- [ ] Native
- [ ] Policy Author

### CODE REVIEW

File: container-images/mds-config/index.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    See the License for the specific language governing permissions and
+    limitations under the License.
+ */
+
+// Express local
+import { ApiServer } from '@mds-core/mds-api-server'
+import { api } from '@mds-core/mds-config'
+import { env } from '@container-images/env-inject'
+
+const { npm_package_name, npm_package_version, npm_package_git_commit, PORT = 4008 } = env()
```
Comment (levi217) [inline container-images/mds-config/index.ts]: identity is using port `4008`. Do we anticipate any port conflicts with `lacuna-mds` and `mds-core` apis?

File: packages/mds-config-service/client.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import fs from 'fs'
+import { promisify } from 'util'
+import { NotFoundError } from '@mds-core/mds-utils'
+
+export const getSettings = async <TSettings extends object>(name = 'settings'): Promise<TSettings> => {
+  const { MDS_CONFIG_PATH = '.' } = process.env
+  const path = MDS_CONFIG_PATH.endsWith('/') ? MDS_CONFIG_PATH : `${MDS_CONFIG_PATH}/`
```
Comment (levi217) [inline packages/mds-config-service/client.ts]: Should it `|| '\'` for Windows locations or does it not matter?

File: packages/mds-config/api.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import express from 'express'
+import { pathsFor, NotFoundError, ServerError } from '@mds-core/mds-utils'
+import client from '@mds-core/mds-config-service'
+import { ConfigApiGetSettingsRequest, ConfigApiResponse } from './types'
+
+function api(app: express.Express): express.Express {
+  app.get(pathsFor('/settings/:name'), async (req: ConfigApiGetSettingsRequest, res: ConfigApiResponse) => {
+    const { name } = req.params
+    try {
+      const settings = await client.getSettings(name)
```
Comment (levi217) [inline packages/mds-config/api.ts]: Is there any security concern using un-sanitized name here? Not sure if `fs.readFile` has any weaknesses around that.

File: container-images/mds-config/index.ts (hunk 1 above)
Comment (drtyh2o) [reply container-images/mds-config/index.ts]: I thought about that. My recommendation is to use a different `PORT` range, perhaps `5000-` for APIs that are not in `mds-core` .. so we can avoid these conflicts.

File: packages/mds-config-service/client.ts (hunk 2 above)
Comment (drtyh2o) [reply packages/mds-config-service/client.ts]: Node.js `path` modules handles all that.

File: packages/mds-config/api.ts (hunk 3 above)
Comment (drtyh2o) [reply packages/mds-config/api.ts]: Not that I am concerned about ... would most likely just return a 404 and the read is scoped to a particular folder.

File: container-images/mds-config/index.ts (hunk 1 above)
Comment (levi217) [reply container-images/mds-config/index.ts]: @invertigo do you have thoughts about this? Using 5xxx in lacuna-mds apis would avoid port conflicts. If you're behind this, I'll try to update identity before it gets deployed.

Comment (twelch) [reply container-images/mds-config/index.ts]: @levi217 I would think lacuna-mds should use a different range.  For apps we use a simple readme to track port assignments and we use 3000-

File: helm/mds/templates/configmap.yaml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: mds-config-files
+  namespace: {{ .Release.Namespace }}
+data:
+{{ (.Files.Glob "mds-config/*.json").AsConfig | indent 2 }}
```
Comment (drtyh2o) [inline helm/mds/templates/configmap.yaml]: @invertigo How to we update this section or the glob pattern to include `.json` and `.json5` extensions.

File: helm/mds/templates/configmap.yaml [hunk 5]
```diff
@@ -4,5 +4,5 @@ metadata:
   name: mds-config-files
   namespace: {{ .Release.Namespace }}
 data:
-{{ (.Files.Glob "files/mds-config/*.json").AsConfig | indent 2 }}
-{{ (.Files.Glob "files/mds-config/*.json5").AsConfig | indent 2 }}
+{{ (.Files.Glob "files/organization-config/*.json").AsConfig | indent 2 }}
+{{ (.Files.Glob "files/organization-config/*.json5").AsConfig | indent 2 }}
```
Comment (drtyh2o) [inline helm/mds/templates/configmap.yaml]: How important is it that these match? I have a minor concern that the concept of an `organization` is not really defined in this context. Maybe `config-settings`?

**Pull request merged**